libs/platform/experience/composition/src/composition.component.ts

```diff
@@ -35,6 +36,26 @@ export class CompositionComponent extends LayoutMixin(
   protected registryService = resolve(ComponentsRegistryService);
   protected layoutBuilder = resolve(LayoutBuilder);
 
+  protected $dynamicVisibilityState = signal(
+    this.contentController.dynamicVisibilityState()
+  );
+
+  @elementEffect()
+  protected toggleVisibilityState = (): void => {
+    if (!this.uid) return;
```

Comment:
should we count for uid for route as well?

Reply:
I don't think so. Route is needed to get components id by route, and then the id is stored to the uid, in result uid only important